### DISCUSSION TITLE
VS Code Plugin

### DISCUSSION THREAD

**Discussion by positlabs** [cables-gl/cables_docs#255] [Feature Requests]
It would be super nice to have some kind of integration with vscode so I can edit scripts in my preferred environment. I'll be able to use whatever linter styles I want, use code formatting, etc. 

It would resolve a lot of issues. Here are the current downsides to the built-in editor, from my perspective: 
- No code format function for custom ops
- Forced code formatting rules for user ops
- Always hitting escape at the wrong time and closing the editor window
- Cables UI crashes and I lose work

Potential upsides to vscode integration: 
- Less tab navigation within cables
- Code snippets for common task
- Scripts saved locally as a backup? 
- Customizable code formatting and linting 

It looks like this is a pretty common use-case for extensions. There is already some tooling built around remote workspaces. https://code.visualstudio.com/api/advanced-topics/remote-extensions

**Comment by wirmachenbunt** [comment]
if this is doable with cables, it sounds like a good idea. as much as i love cables, the code editor is not much fun when you come from VS. not that i'm using all the features of VS but some are really helpful. like code formating, magic code completion, automatic scaffolding of if, foreach, classes etc.

so yeah, happy to have it. probably a pro feature for a few ?
